File: running-app-backend/product/models/product.py
import uuid

from django.db import models 
from django.core.validators import MaxLengthValidator

class Product(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, db_index=True)
    name = models.CharField(max_length=100)
    brand = models.ForeignKey(
        "product.Brand", related_name="brand", on_delete=models.CASCADE)
    category = models.ForeignKey(
        "product.Category", related_name="category", on_delete=models.CASCADE)
    price = models.IntegerField()
    description = models.TextField(
        blank=True, 
        null=True,
        validators=[MaxLengthValidator(255, 'The field can contain at most 200 characters')]
    )
    uploaded_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    def __str__(self):
        return f"{self.name} {self.brand} {self.category}"
    
class ProductImage(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, db_index=True)
    product = models.ForeignKey("product.Product", related_name="product_image", on_delete=models.CASCADE)
    image = models.ImageField(upload_to="", default="")
    
    def __str__(self):
        return f"{self.product.name} {self.image}"

File: running_app/lib/models/product/product.dart
import 'package:running_app/models/product/brand.dart';

class Product {
  final String? id;
  final String? name;
  final Brand? brand;
  final int? price;

  Product({
    required this.id,
    required this.name,
    required this.brand,
    required this.price
  });

  Product.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        brand = Brand.fromJson(json['brand']),
        price = json['price'];

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'brand': brand,
      'price': price,
    };
  }

  @override
  String toString() {
    return 'Product{id: $id, name: $name, brand: $brand, price: $price}';
  }
}

class DetailProduct extends Product {
  final Map<String, dynamic>? category;
  final String? description;
  final String? uploadedAt;
  final String? updatedAt;

  DetailProduct({
    String? id,
    String? name,
    Brand? brand,
    int? price,
    required this.category,
    required this.description,
    required this.uploadedAt,
    required this.updatedAt,
  }) : super(
    id: id,
    name: name,
    brand: brand,
    price: price
  );

  DetailProduct.fromJson(Map<String, dynamic> json)
      : category = json['category'],
        description = json['description'],
        uploadedAt = json['uploaded_at'],
        updatedAt = json['updated_at'],
        super(
          id: json['id'],
          name: json['name'],
          brand: json['brand'],
          price: json['price']
        );

  @override
  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = super.toJson();
    data['category'] = category;
    data['description'] = description;
    data['uploaded_at'] = uploadedAt;
    data['updated_at'] = updatedAt;
    return data;
  }

  @override
  String toString() {
    return 'DetailProduct{${super.toString()}, category: $category, description: $description, uploadedAt: $uploadedAt, updatedAt: $updatedAt}';
  }
}